### PR TITLE
Upgrade to repo-config v0.7.5 with regenerated templates

### PR DESCRIPTION
**Pull request by noah-kreutzer-frequenz**
with manual removals according to upgrades from v0.6.2 to v0.7.5
s/o to cwasicki

### CODE REVIEW

**Comment by shsms** [comment]
@llucax could you also please check if @noah-kreutzer-frequenz is missing some permissions?  Because CI doesn't automatically run, but requires approval.

File: .github/ISSUE_TEMPLATE/bug.yml
Comment (llucax) [inline .github/ISSUE_TEMPLATE/bug.yml]: General comments about all TODOs added here: You should probably address them and remove them. Since this is a preexisting repository, it is very likely you want to remove them without much thought, as the setup should have been already done.

We should have a way to determine if this is a first time generation or not and only put `TODOs` for first time generation, but for now it needs to be manually cleaned up on every update :disappointed: 

* https://github.com/frequenz-floss/frequenz-repo-config-python/issues/186

File: .github/containers/nox-cross-arch/arm64-ubuntu-20.04-python-3.11.Dockerfile
Comment (llucax) [inline .github/containers/nox-cross-arch/arm64-ubuntu-20.04-python-3.11.Dockerfile:1]: We are [still deciding](https://github.com/frequenz-floss/frequenz-repo-config-python/discussions/177) if we want to test for arm64, and how, and in which type of repo. For now I would recommend removing all the files related to arm64 testing. I will comment on everything that needs removing regarding arm64.

We should also have a repo-config option to avoid generating arm64 testing to avoid having to clean up all this over and over again when re-generating templates:

* https://github.com/frequenz-floss/frequenz-repo-config-python/issues/187

File: .github/containers/nox-cross-arch/entrypoint.bash
Comment (llucax) [inline .github/containers/nox-cross-arch/entrypoint.bash:1]: This file is amr64-related, it should be removed.

File: .github/workflows/ci.yaml
Comment (llucax) [inline .github/workflows/ci.yaml]: This is amr64-related, it should be removed.

```suggestion
```

File: .github/workflows/ci.yaml
Comment (llucax) [inline .github/workflows/ci.yaml]: This is amr64-related, it should be changed.
```suggestion
    needs: ["nox-all", "test-installation", "protolint"]
```

File: CONTRIBUTING.md
Comment (llucax) [inline CONTRIBUTING.md]: This is amr64-related, it should be removed.
```suggestion
```

File: .gitignore
Comment (llucax) [inline .gitignore]: Mmm, interesting that this got removed, I think it should stay, because these files will be generated as soon as you use `pip install -e .`, and you don't want to add them to the repo accidentally. We should add this too to repo-config:

* https://github.com/frequenz-floss/frequenz-repo-config-python/issues/185

File: .github/workflows/ci.yaml
Comment (llucax) [inline .github/workflows/ci.yaml:173]: Just FYI, this is arm64-realted, but it only tests that the package can be installed, is more lightweight than the full testing in arm64 and it runs early in the workflow and as a parallel step, so it usually doesn't affect the total workflow run time, so I would leave it.

File: README.md
Comment (llucax) [inline README.md]: All of this **should not** be removed!

File: RELEASE_NOTES.md
Comment (llucax) [inline RELEASE_NOTES.md]: This probably should not be removed either.

File: noxfile.py
Comment (llucax) [inline noxfile.py]: This should not be updated.
```suggestion
# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
```

File: proto/frequenz/api/weather/weather.proto
Comment (llucax) [inline proto/frequenz/api/weather/weather.proto:1]: This file also **should not** be updated!

File: py/frequenz/api/weather/__init__.py
Comment (llucax) [inline py/frequenz/api/weather/__init__.py:1]: This file also **should not** be updated!

File: pyproject.toml
Comment (llucax) [inline pyproject.toml]: There is a but in repo-config v0.7.5, it is using an older release, this should be 0.7.5.
```suggestion
  "frequenz-repo-config[api] == 0.7.5",
```

File: pyproject.toml
Comment (llucax) [inline pyproject.toml]: You probably don't want to downgrade this.
```suggestion
  "frequenz-api-common >= 0.5.0, < 0.6.0",
```

File: pyproject.toml
Comment (llucax) [inline pyproject.toml]: You probably don't want to downgrade `mkdocs-material`, and actually you probably don't want to downgrade **any** dependency.

File: pyproject.toml
Comment (llucax) [inline pyproject.toml]:
```suggestion
  "frequenz-repo-config[api] == 0.7.5",
```

File: pyproject.toml
Comment (llucax) [inline pyproject.toml]:
```suggestion
  "frequenz-repo-config[api] == 0.7.5",
```

File: pyproject.toml
Comment (llucax) [inline pyproject.toml]:
```suggestion
  "frequenz-repo-config[extra-lint-examples] == 0.7.5",
```

File: mkdocs.yml
Comment (llucax) [inline mkdocs.yml:23]: You need to remove the old directory `docs/overrides` too, please have a look at the [updating notes in the releases notes](https://github.com/frequenz-floss/frequenz-repo-config-python/releases) (for all the versions you are jumping too).

File: mkdocs.yml
Comment (llucax) [inline mkdocs.yml:17]: You also need to remove the old logo file in `docs/logo.png`.

File: mkdocs.yml
Comment (llucax) [inline mkdocs.yml:61]: Same here, you need to remove the old directory `docs/css` and its files.

**Review by llucax — COMMENTED** [review]
I'm really sorry but at the moment updating the templates in repo-config need a lot of manual work. There is a lot that needs to be changed (or not changed) :disappointed:

File: .github/ISSUE_TEMPLATE/bug.yml
Comment (cwasicki) [reply .github/ISSUE_TEMPLATE/bug.yml]: I recommended to leave them in since they will come back anyway :) But I realize they could cause problems with nox, right?

File: .github/containers/nox-cross-arch/entrypoint.bash
Comment (noah-kreutzer-frequenz) [reply .github/containers/nox-cross-arch/entrypoint.bash:1]: done.

File: .github/containers/nox-cross-arch/arm64-ubuntu-20.04-python-3.11.Dockerfile
Comment (noah-kreutzer-frequenz) [reply .github/containers/nox-cross-arch/arm64-ubuntu-20.04-python-3.11.Dockerfile:1]: does this mean this whole file should be deleted? @llucax

File: .github/ISSUE_TEMPLATE/bug.yml
Comment (llucax) [reply .github/ISSUE_TEMPLATE/bug.yml]: For the python stuff yes, pylint will complain about TODOs

File: .github/containers/nox-cross-arch/arm64-ubuntu-20.04-python-3.11.Dockerfile
Comment (llucax) [reply .github/containers/nox-cross-arch/arm64-ubuntu-20.04-python-3.11.Dockerfile:1]: Yes

File: CONTRIBUTING.md
Comment (llucax) [inline CONTRIBUTING.md:1]: Not sure why you are adding and removing empty lines in this file...

File: .github/ISSUE_TEMPLATE/bug.yml
Comment (llucax) [reply .github/ISSUE_TEMPLATE/bug.yml]: I'm not sure if you are leaving some TODOs intentionally, I've seen that you removed some, but some other are still here.

File: mkdocs.yml
Comment (llucax) [inline mkdocs.yml]: In this file there are also a lot of TODOs left.

File: pyproject.toml
Comment (llucax) [reply pyproject.toml]: This one still needs to be updated

**Review by llucax — COMMENTED** [review]
Just a few things missing.

Also, the commit messages are not super useful, not sure if you plan to squash most of it together before merging. If you haven't yet, I recommend having a look at: https://github.com/frequenz-io/rnd/blob/master/doc/coding-standards/commit-messages.md

File: CONTRIBUTING.md
Comment (noah-kreutzer-frequenz) [reply CONTRIBUTING.md:1]: oh, I actually didin't do it on purpose. Don't know what happend here, sorry

File: CONTRIBUTING.md
Comment (llucax) [inline CONTRIBUTING.md:200]: Weird, I can't see what changed here. :thinking:

**Review by llucax — APPROVED** [review]

**Comment by llucax** [comment]
Since there are no visible changes to the user, you can add the label `cmd:skip-release-notes` to make the check `Release Notes Check` pass, I now added the label myself.

**Comment by llucax** [comment]
Since there are no visible changes to the user, you can add the label `cmd:skip-release-notes` to make the check `Release Notes Check` pass, I now added the label myself.

**Comment by noah-kreutzer-frequenz** [comment]
I have merged my commit into one I hope, it's fine that way.
Since this is my first official PR I would be very grateful to recieve some further feedback @llucax @shsms

**Review by shsms — APPROVED** [review]

**Review by llucax — APPROVED** [review]
LGTM, only 2 minor comments:

1. Commit messages should be wrapped at 80 columns
2. I would have squashed the last 2 commits together, just having one commit to undo half of it in the following commit makes reviewing much harder

**Comment by noah-kreutzer-frequenz** [comment]
Alright I'll have that in mind for the next time.
Thanks for your tipps and patience @shsms @llucax @cwasicki 🙏🏾

**Pull request merged**
